balancer/weightedtarget/weightedtarget_test.go

```diff
@@ -199,15 +199,15 @@ func (s) TestWeightedTarget(t *testing.T) {
 
 	// Send subconn state change.
 	sc1 := <-cc.NewSubConnCh
-	wtb.UpdateSubConnState(sc1, balancer.SubConnState{ConnectivityState: connectivity.Connecting})
+	sc1.UpdateState(balancer.SubConnState{ConnectivityState: connectivity.Connecting})
 	<-cc.NewPickerCh
-	wtb.UpdateSubConnState(sc1, balancer.SubConnState{ConnectivityState: connectivity.Ready})
+	sc1.UpdateState(balancer.SubConnState{ConnectivityState: connectivity.Ready})
 	p := <-cc.NewPickerCh
 
 	// Test pick with one backend.
 	for i := 0; i < 5; i++ {
 		gotSCSt, _ := p.Pick(balancer.PickInfo{})
-		if !cmp.Equal(gotSCSt.SubConn, sc1, cmp.AllowUnexported(testutils.TestSubConn{})) {
+		if gotSCSt.SubConn != sc1 {
```

Comment:
Are we able to do direct struct comparisons instead of a `cmp.Equal` because you switched the channels in the `TestClientConn` to return the concrete subConn instead of the interface or any other reason?

Reply:
Yes, I believe that's the reason this now works.  Also `cmp.Equal` stopped working because of the `stateListener` and rather than updating every one of these to handle "one more special thing" in the state being compared, I thought a simpler, direct comparison would be better.